exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServer.java

```diff
@@ -190,7 +198,15 @@ public void handle(HttpExchange exchange) throws IOException {
         } else {
           out = exchange.getResponseBody();
         }
-        serializer.write(metrics, out);
+        HashSet<String> conflictHeaderNames = serializer.write(metrics, out);
```

Comment:
```suggestion
        Set<String> conflictHeaderNames = serializer.write(metrics, out);
```

Reply:
Was using `HashSet` in these places because the caller mutates the set to find any new conflict names that haven't produced a warning yet. Probably better to stick to the book, returning `Set` and having the caller make a defensive copy.

Reply:
Plain Set is mutable by its API, no?

Reply:
Yes, but the immutable varieties still implement `Set` and throw when mutating. I'll just switch to `Set` though - there's unit tests that confirm the mutability.